android/src/main/java/com/dataxad/flutter_mailer/FlutterMailerPlugin.java

```diff
@@ -0,0 +1,293 @@
+package com.dataxad.flutter_mailer;
+
+import androidx.annotation.NonNull;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.core.content.FileProvider;
+
+import android.text.Html;
+import android.text.Spanned;
+import android.util.Log;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.PluginRegistry;
+
+/**
+ * FlutterMailerPlugin
+ */
+public class FlutterMailerPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware, PluginRegistry.ActivityResultListener {
```

Comment:
Just a idea: Because the mailer plugin is fairly complex, consider moving the MethodCallHandler into a separate class `MethodCallHandlerImpl`. It makes it much clearer where Flutter lifecycle begins & ends.

Reply:
good idea, but i dont believe i will be doing so in this PR.

Reply:
decided to try to implement it any way. can you please have another look. also published to dart pad as a dev version if you want to test it out.
`0.5.0-dev.3`